go.mod

```diff
@@ -3,6 +3,9 @@ module github.com/siderolabs/omni
 go 1.23.3
 
 replace (
+	// todo: remove when merged
+	github.com/siderolabs/omni-infra-provider-bare-metal v0.0.0-20241014092637-c3e892b87244 => github.com/utkuozdemir/sidero-omni-infra-provider-bare-metal v0.0.0-20241021213554-cc93e594a0d2
```

Comment:
That's a good point, and actually it was unused. The initial idea and the reason was to import the GRPC server API types that the provider serves, so that Omni could send commands to the machines (e.g. Reboot) over the reverse GRPC tunnel, `Omni->Provider->Machine`.

I initially did that, but then removed to keep the scope smaller in this PR.

So I now removed this import, and also decided to remove the whole reverse tunnel logic. I'll keep that in a separate Draft PR for now, as I'm having second thoughts on whether it is necessary at all. It would only be used for reboot for the time being, and imo it is perfectly doable over the resources at hand as well (similar to wipeID, something like rebootID).

So I'll discuss this with @Unix4ever when he is back :)